egress/src/filter.rs

```diff
@@ -0,0 +1,1415 @@
+                    .unwrap()
+                    .0
+                    .from;
+                let last_time = time_pri_typ_ts
+                    .iter()
+                    .min_by_key(|item| (item.0.to))
+                    .unwrap()
+                    .0
+                    .to;
+                // keep looping until no more holes...
+                filter.insert(
+                    label,
+                    fill_holes(
+                        time_pri_typ_ts,
+                        Timerange {
+                            from: first_time,
+                            to: last_time,
+                        },
+                    ),
+                );
+            }
+        }
+    }
+    // sort by descending from time since otherwise unordered
+    // removing this will cause tests to fail... if its ok for other stuff could potentially be moved into the test framework somehow?
+    for list in filter.values_mut() {
+        list.sort_by_key(|item| (item.from));
+    }
+    eprintln!("length filter: {:?}", filter.len());
+    Ok(filter)
+}
+
+pub async fn get_filter(
+    conn: &PooledPgConn<'_>,
+    from: DateTime<Utc>,
+    to: DateTime<Utc>,
+    filter_table: FilterTimeseriesTableLock,
+    label: FilterLabel,
+) -> Result<Option<Vec<FilterData>>, tokio_postgres::Error> {
+    // get the background filter list, and lookup this label
+    // have to clone, otherwise does not work...
+    let ft = filter_table
+        .read()
+        .map_err(|e| Error::Lock(e.to_string()))
+        .unwrap()
+        .clone();
+
+    let filter = ft.get(&label);
+    // TODO: if the label has none for sensor / level should it match on all???
+    // create a structure to keep what is applicable
+    let mut applicable_ts: Vec<(i64, DateTime<Utc>, DateTime<Utc>)> = vec![];
+    // fill the structure
+    match filter {
+        Some(options) => {
+            for opt in options {
+                //    // is this applicable?
+                let ft_t = timerange_overlap(
+                    Timerange {
+                        from: Some(from),
+                        to: Some(to),
+                    },
+                    Timerange {
+                        from: Some(opt.from),
+                        to: opt.to,
+                    },
+                );
+                // have overlap
+                if let Some(times) = ft_t {
+                    if let Some(tt) = times.to {
+                        applicable_ts.push((opt.tsid, times.from.unwrap(), tt));
+                    } else {
+                        // open ended (to time from request)
+                        applicable_ts.push((opt.tsid, times.from.unwrap(), to));
+                    }
+                }
+            }
+            if applicable_ts.is_empty() {
+                return Ok(None); // no applicable time series for this time range?
+            }
+        }
+        None => return Ok(None), // no prioritized timeseries
```

Comment:
Should these return `Ok`? Is it better for them to return an error, since the label/timeseries cannot be found? And under which conditions would these apply?

Reply:
Returning error "no applicable timeseries found for this time range"